Tools/wasm/wasm_assets.py

```diff
@@ -105,34 +122,59 @@
 )
 
 
-OMIT_ABSOLUTE = {SRCDIR_LIB / name for name in OMIT_FILES}
-OMIT_SUBDIRS_ABSOLUTE = tuple(str(SRCDIR_LIB / name) for name in OMIT_SUBDIRS)
-
-
-def filterfunc(name: str) -> bool:
-    return not name.startswith(OMIT_SUBDIRS_ABSOLUTE)
-
-
 def create_stdlib_zip(
-    args: argparse.Namespace, compression: int = zipfile.ZIP_DEFLATED, *, optimize: int = 0
+    args: argparse.Namespace,
+    *,
+    optimize: int = 0,
 ) -> None:
-    sysconfig_data = list(args.builddir.glob(SYSCONFIGDATA_GLOB))
-    if not sysconfig_data:
-        raise ValueError("No sysconfigdata file found")
+    def filterfunc(name: str) -> bool:
+        return not name.startswith(args.omit_subdirs_absolute)
 
     with zipfile.PyZipFile(
-        args.wasm_stdlib_zip, mode="w", compression=compression, optimize=0
+        args.wasm_stdlib_zip, mode="w", compression=args.compression, optimize=optimize
     ) as pzf:
+        if args.compresslevel is not None:
+            pzf.compresslevel = args.compresslevel
+        pzf.writepy(args.sysconfig_data)
         for entry in sorted(args.srcdir_lib.iterdir()):
             if entry.name == "__pycache__":
                 continue
-            if entry in OMIT_ABSOLUTE:
+            if entry in args.omit_files_absolute:
                 continue
             if entry.name.endswith(".py") or entry.is_dir():
                 # writepy() writes .pyc files (bytecode).
                 pzf.writepy(entry, filterfunc=filterfunc)
-        for entry in sysconfig_data:
-            pzf.writepy(entry)
+
+
+def detect_extension_modules(args: argparse.Namespace):
+    modules = {}
+
+    # disabled by Modules/Setup.local ?
+    with open(args.builddir / "Makefile") as f:
+        for line in f:
+            if line.startswith("MODDISABLED_NAMES="):
+                disabled = line.split("=", 1)[1].strip().split()
+                for modname in disabled:
+                    modules[modname] = False
```

Comment:
Any reason this isn't a set?

Reply:
I have to track disabled and later enabled modules. A dict seemed easier to use than a set.